geoutils/raster/raster.py

```diff
@@ -2464,15 +2480,29 @@ def icrop(
         if len(bbox) != 4:
             raise ValueError("bbox must be a list or tuple of four integers: (xmin, ymin, xmax, ymax).")
 
-        # Convert pixel coordinates to georeferenced coordinates using the transform
```

Comment:
The argument description above was ambiguous for me:

>         :param bbox: Pixel-based bounding box as (xmin, ymin, xmax, ymax) in pixel coordinates.

Maybe similar language as in the function description "Bounding box based on indices of the raster array".

We could also add a note in the general description to clarify which this is strikingly different, something like: "Note that, by convention, a raster array has the Y axis as first in the array, and that this Y axis is also flipped (first index corresponds to the highest-value coordinate)."